### PR TITLE
doc: Correct documented default timeout to 40s

### LINKED ISSUE #496: Doc - timeout does no longer default to 1 minute
This is a kind of follow-up issue on: https://github.com/elastic/terraform-provider-ec/pull/128

## Readiness Checklist

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior

[doc](https://github.com/elastic/terraform-provider-ec/blob/v0.4.1/docs/index.md?plain=1#L102) should follow the [code](
https://github.com/elastic/terraform-provider-ec/blob/v0.4.1/ec/provider.go#L59).

## Current Behavior

[doc](https://github.com/elastic/terraform-provider-ec/blob/v0.4.1/docs/index.md?plain=1#L102) says `Defaults to 1 minute` but [code](
https://github.com/elastic/terraform-provider-ec/blob/v0.4.1/ec/provider.go#L59) says `defaultTimeout = 40 * time.Second`

## Terraform definition

n/a

## Steps to Reproduce

n/a

## Context

The inconsistency makes me a bit confused. That's all.

## Possible Solution

Update the [doc](https://github.com/elastic/terraform-provider-ec/blob/v0.4.1/docs/index.md?plain=1#L102) with `Defaults to 40 seconds ("40s")`.

## Your Environment

* Version used: 
* Running against Elastic Cloud SaaS or Elastic Cloud Enterprise and version:
* Environment name and version (e.g. Go 1.9):
* Server type and version:
* Operating System and version:
* Link to your project:

### PR DESCRIPTION
**Pull request by sakurai-youhei**
Correct the default timeout in documentation according to the change in #128.

Closes #496

## Description

In #128, the default timeout was changed to 40 seconds [here](https://github.com/elastic/terraform-provider-ec/pull/128/files#diff-8a99a7c023fb0016302c46546d8bb5ae8bc57df60f6ee46166faf4d536fceff4R57).

This PR makes the doc up to date.

## Related Issues

https://github.com/elastic/terraform-provider-ec/issues/496

## Motivation and Context

Just for better consistency between code and doc.

## How Has This Been Tested?

Just by the workflow in Github Actions.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Readiness Checklist

- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ In addition to [generic `provider` arguments](https://www.terraform.io/docs/conf
   targeting ESS.
 
 * `timeout` - (Optional) This setting allows the user to set a custom timeout in the
-  individual HTTP request level. Defaults to 1 minute (`"1m"`), but might need to be tweaked if timeouts
+  individual HTTP request level. Defaults to 40 seconds (`"40s"`), but might need to be tweaked if timeouts
```
Comment (nrichers) [inline docs/index.md:102]:
```suggestion
  individual HTTP request level. Defaults to 40 seconds (`"40s"`), but might need to be adjusted if timeouts
```

**Review by nrichers — COMMENTED** [review]
Minor edit you can suggest directly.

File: docs/index.md (hunk 1 above)
Comment (sakurai-youhei) [reply docs/index.md]: Thanks a lot!

**Review by nrichers — APPROVED** [review]
LGTM! 🚢

**Review by tobio — APPROVED** [review]

**Pull request merged**